### PR TITLE
Cleaned up and modified global variables documentation

### PR DESCRIPTION
**Pull request by Denton-L**

### CODE REVIEW

File: docs/units-and-global-variables.rst [hunk 1]
```diff
@@ -111,6 +111,21 @@ same as ``sha3(uint16(0x12))``.
 
 It might be that you run into Out-of-Gas for ``sha256``, ``ripemd160`` or ``ecrecover`` on a *private blockchain*. The reason for this is that those are implemented as so-called precompiled contracts and these contracts only really exist after they received the first message (although their contract code is hardcoded). Messages to non-existing contracts are more expensive and thus the execution runs into an Out-of-Gas error. A workaround for this problem is to first send e.g. 1 Wei to each of the contracts before you use them in your actual contracts. This is not an issue on the official or test net.
 
+.. index:: send, balance
+
+Address Related
+---------------
+
+``<address>.balance`` (``uint256``):
```
Comment (chriseth) [inline docs/units-and-global-variables.rst:119]: Please add links to the address type here and from the address type to here, so that we do not forget updating both if something changes.

File: docs/units-and-global-variables.rst [hunk 2]
```diff
@@ -111,6 +111,21 @@ same as ``sha3(uint16(0x12))``.
...
+    balance of the address in Wei
+``<address>.send(uint256 amount) returns (bool)``:
+    send given amount of Wei to address, returns ``false`` on failure
+
+.. warning::
+    Since it is possible for calls to ``send`` to fail, such as
```
Comment (chriseth) [inline docs/units-and-global-variables.rst]: I extended that warning in a pending pull request, could you try to combine both?

File: docs/miscellaneous.rst [hunk 3]
```diff
@@ -148,6 +148,24 @@ Pitfalls
 Unfortunately, there are some subtleties the compiler does not yet warn you about.
 
 - In ``for (var i = 0; i < arrayName.length; i++) { ... }``, the type of ``i`` will be ``uint8``, because this is the smallest type that is required to hold the value ``0``. If the array has more than 255 elements, the loop will not terminate.
+- If a contract receives Ether (without a function being called), the fallback function is executed. The contract can only rely
+  on the "gas stipend" (2300 gas) being available to it at that time. This stipend is not enough to access storage in any way.
+  To be sure that your contract can receive Ether in that way, check the gas requirements of the fallback function.
+- If you want to send ether using ``address.send``, there are certain details to be aware of:
+  1. If the recipient is a contract, it causes its fallback function to be executed which can in turn call back into the sending contract
+  2. Sending Ether can fail due to the call depth going above 1024. Since the caller is in total control of the call
```
Comment (Denton-L) [inline docs/miscellaneous.rst:157]: What is the maximum call depth? Is it >1024 or >= 1024? This line, and the line mentioning 1023, makes this a little confusing.

File: docs/types.rst [hunk 4]
```diff
@@ -84,6 +86,12 @@ and to send Ether (in units of wei) to an address using the ``send`` function:
 .. note::
     If ``x`` is a contract address, its code (more specifically: its fallback function, if present) will be executed together with the ``send`` call (this is a limitation of the EVM and cannot be prevented). If that execution runs out of gas or fails in any way, the Ether transfer will be reverted. In this case, ``send`` returns ``false``.
 
+.. warning::
+    There are some dangers in using ``send``: The transfer fails if the call stack depth is at 1023
```
Comment (Denton-L) [inline docs/types.rst]: See the other thing about 1024.

File: docs/miscellaneous.rst (hunk 3 above)
Comment (chriseth) [inline docs/miscellaneous.rst:157]: The maximum depth is 1024, but it is not clear whether the first call is at depth 0 or at depth 1 :-)

File: docs/types.rst (hunk 4 above)
Comment (chriseth) [inline docs/types.rst]: Ok, make this 1024

File: docs/units-and-global-variables.rst [hunk 5]
```diff
@@ -121,10 +121,13 @@ Address Related
 ``<address>.send(uint256 amount) returns (bool)``:
     send given amount of Wei to address, returns ``false`` on failure
 
+For more information, see the section on :ref:`address`.
+
 .. warning::
-    Since it is possible for calls to ``send`` to fail, such as
-    if there are insufficient funds, it is good practice to
-    check the return value whenever calling ``send``.
+    There are some dangers in using ``send``: The transfer fails if the call stack depth is at 1023
```
Comment (chriseth) [inline docs/units-and-global-variables.rst]: also change to 1024 please

**Comment by Denton-L** [comment]
Updated. Also, it actually passes the Travis CI build now.

File: docs/units-and-global-variables.rst [hunk 6]
```diff
@@ -111,20 +111,18 @@ same as ``sha3(uint16(0x12))``.
...
 Address Related
 ---------------
 
 ``<address>.balance`` (``uint256``):
-    balance of the address in Wei
+    balance of the :ref:`address` in Wei
 ``<address>.send(uint256 amount) returns (bool)``:
-    send given amount of Wei to address, returns ``false`` on failure
-
-For more information, see the section on :ref:`address`.
```
Comment (chriseth) [inline docs/units-and-global-variables.rst]: Why did you delete this line?

Comment (Denton-L) [inline docs/units-and-global-variables.rst]: I incorporated the links into the lines above, in the description. This is more in line with how it was done in other sections, like the one right below it.

Comment (chriseth) [inline docs/units-and-global-variables.rst]: But now it does not say anymore that more information can be found behind that link. Shouldn't we mention that?

**Comment by Denton-L** [comment]
Alright, added that section back in.
